### ISSUE TITLE
Cocoapods 1.11.0 throwing error "userinfo not supported.  [RFC3986]" when trying to install a private pod.

### ISSUE THREAD

**Issue by nahuelDeveloper** [CocoaPods/CocoaPods#10941]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Updated CocoaPods to 1.11.0. Run `pod install` to install dependencies in my project. We have a private pod, so i pass two environment variables for that particular pod's url, and also for the Specs url. The Specs url is similar to the following url:

'https://' + ENV['USERNAME'] + ':' + ENV['CREDENTIALS'] + '@bitbucket.org/foobar/Specs.git'

## What did you expect to happen?

Install all pod dependencies correctly. Installation was working fine when using CocoaPods 1.10.2

## What happened instead?

The installation failed, with the following error :
Couldn't determine repo type for URL: `https://foo:***@bitbucket.org/foobar/Specs.git`: userinfo not supported.  [RFC3986]

## CocoaPods Environment

   CocoaPods : 1.11.0
        Ruby : ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin20]
    RubyGems : 3.0.3
        Host : macOS 11.5.1 (20G80)
       Xcode : 12.5.1 (12E507)
         Git : git version 2.30.1 (Apple Git-130)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib

## Project that demonstrates the issue

I don't have a project that i can link, the pod that causes this issue is private and requires credentials to be installed.

I'm creating this issue because i think it could be a bug that is related to the latest release of CocoaPods, but i'm not sure. Maybe there was an update and the RCF 3986 specification is no longer supported? If that's the case, please let me know and i will search for an alternative.

I searched for similar issues, and i also read the Changelog to see the changes of the latest release, but i couldn't find anything. Any help will be very appreciated, thanks in advance!

**Comment by dnkoutso** [comment]
This is fixed in Core by https://github.com/CocoaPods/Core/issues/705 and the 1-11-stable branch CHANGELOG includes it. We have not merged it yet to master.

Lets close for now and we can verify once 1.11.1 ships.

**Comment by dnkoutso** [comment]
@nahuelDeveloper can you point to the latest `1-11-stable` branch to try it out and ensure it works?

**Comment by dnkoutso** [comment]
it seems that we can still parse this URL even without the https://github.com/CocoaPods/Core/issues/705 fix....... so this might be something different...?

**Comment by dnkoutso** [comment]
@nahuelDeveloper can you also try bumping to a version of Ruby like 2.6.3 or 2.6.6?

**Comment by dnkoutso** [comment]
ah I can now reproduce, this is a different issue actually:

```
[!] Couldn't determine repo type for URL: `https://foo:bar@bitbucket.org/foobar/Specs.git`: userinfo not supported.  [RFC3986]
```

Has to do with `OpenURI`.

**Comment by dnkoutso** [comment]
Looks like related to https://github.com/CocoaPods/CocoaPods/commit/4a9a504529ca11340de0cfa5d5be2dcb82d09328

**Comment by dnkoutso** [comment]
@jasl any ideas?

**Comment by dnkoutso** [comment]
I think I got a fix.

**Comment by dnkoutso** [comment]
Fixed by https://github.com/CocoaPods/CocoaPods/pull/10942

**Comment by nahuelDeveloper** [comment]
@dnkoutso thank you very much for the fix! I ran pod install again today with the latest release of CocoaPods (1.11.1), and i'm grateful to see the "Pod installation complete!" message again.

**Comment by dnkoutso** [comment]
excellent! also shipped 1.11.2 with an additional fix if you want.
